posts/2020-11-20-JNDI-gRPC.adoc

```diff
@@ -0,0 +1,361 @@
+      </ejb-jar-bnd>
+   </ejbApplication>
+----
+
+[customibm2]
+===== Specify Custom bindings in ibm-ejb-jar-bnd.xmi for EJB 1.x/2.x
+
+Following is an example of how to configure custom bindings for EJB 1.x or 2.x beans in an EJB JAR module in `ibm-ejb-jar-bnd.xmi`
+
+EJB 1.x and 2.x provide a single JNDI name that applies to both the remote and local home:
+
+
+[source, XML]
+----
+   <ejbBindings xmi:id="BeanBinding_8" jndiName="suite/r6x/base/misc/poollimits/SLCMTTxTimeoutHome">
+      <enterpriseBean xmi:type="ejb:Session" href="META-INF/ejb-jar.xml#SLCMTTxTimeout"/>
+   </ejbBindings>
+----
+
+For a bean with both a remote and local home, the above will provide the following custom bindings:
+
+
+[source, text]
+----
+   Remote Home : suite/r6x/base/misc/poollimits/SLCMTTxTimeoutHome
+   Local Home  : local:suite/r6x/base/misc/poollimits/SLCMTTxTimeoutHome
+----
+
+==== Feature configuration details
+
+Support for custom and legacy default JNDI names is enabled by default for all Enterprise Bean (EJB) features. This support will not interfere with the existing specification defined `java`: support. However, it is possible to completely disable the new support with the following setting in server.xml:
+
+
+[source, XML]
+----
+    <ejbContainer bindToServerRoot="false"/>
+----
+
+It is also possible to disable just the legacy short form default JNDI name support (i.e. the bean is bound using the interface name) with the following setting in server.xml:
+
+
+[source, XML]
+----
+   <ejbContainer disableShortDefaultBindings="true"/>
+----
+
+Since the new support for customer JNDI names and legacy defaults provider alternative JNDI names, it is now possible to disable the EJB specification required JNDI names. This is done in server.xml as follows:
+
+
+[source, XML]
+----
+   <ejbContainer bindToJavaGlobal="false"/>
+----
+
+Finally, the following new configuration attribute on the `<ejbContainer>` element in open-liberty enables the WebSphere traditional behavior of failing application start when multiple beans are bound to the same JNDI name:
+
+[source, XML]
+----
+    <ejbContainer customBindingsOnError="FAIL"/>
+----
+
+
+[#grpc]
+== Provide and consume gRPC services from your web applications
+
+link:https://grpc.io/docs/what-is-grpc/introduction/[gRPC] is a high-performance, open source universal RPC framework. gRPC support on Liberty allows developers to both provide and consume gRPC services from your web applications. The introduction of gRPC support in Open Liberty now means you can take advantage of the benefits of gRPC more easily than before. Those benefits include great performance, simple service definitions via Protocol Buffers, cross-platform and language support, and wide industry adoption.
+
+These two features which were previously in our beta are now available in this months GA: `grpc-1.0`, which enables gRPC services, and `grpcClient-1.0`, which enables the use of a gRPC client for outbound calls.
```

Comment:
This seems like a run-on sentence - maybe something like:
```suggestion
These two features were previously only available beta are now generally available: 
- `grpc-1.0`, which enables gRPC services, and
- `grpcClient-1.0`, which enables the use of a gRPC client for outbound calls.
```

Reply:
Done